drjt/MyDate/lib/MyDate.pm

```diff
@@ -0,0 +1,169 @@
+package MyDate;
+
+use 5.010;
+use strict;
+use warnings;
+
+use Carp qw( croak );
+use Time::Piece;
+
+=head1 NAME
+
+MyDate - The great new MyDate!
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+
+=head1 SYNOPSIS
+
+Quick summary of what the module does.
+
+Perhaps a little code snippet.
+
+    use MyDate;
+
+    my $foo = MyDate->new();
+    ...
+
+=head1 EXPORT
+
+A list of functions that can be exported.  You can delete this section
+if you don't export anything, such as for a purely object-oriented module.
+
+=head1 SUBROUTINES/METHODS
+
+=head2 AUTOLOAD
+
+=cut
+
+sub AUTOLOAD {
+  our $AUTOLOAD;
+  my $method = $AUTOLOAD =~ s/.*:://r;
+  if ( $method eq 'date' ) {
```

Comment:
Splitting your functionality over this control structure means you have to write a function for each attribute you need. Since you're defining named functions for each attribute, you'd be better off leaving out all the AUTOLOAD stuff as you're gaining no benefit from it this way.